app/controllers/users_controller.rb

```diff
@@ -1,16 +1,27 @@
 class UsersController < ApplicationController
+  include SocialAccountsHelper
   before_action :authorize, except: %i[show index]
+  helper_method :generate_link,  :get_supported_social_networks
+
 
   def show
     @user = User.find(params[:id])
   end
 
-  def edit; end
+  def edit
+    @user = User.find(params[:id])
+    # prototype for create social account form
+    @social_account = @user.social_accounts.build #
+  end
 
   def update
-    return render :edit unless @user.update(user_params)
-
-    redirect_to @user
+    # prototype for create social account form
+    if @user.update(user_params)
```

Comment:
this can be simplified to `return redirect_to @user if @user.update(user_params)`